### PR TITLE
add docs for k8s kubeconfig, defaultNamespace

### LINKED ISSUE #3669: [1.12] Update docs for Kubernetes components
All Kubernetes components (binding, secret store, crypto) have a new metadata option "kubeconfigPath". Also, the secret store now has a metadata option (not required) "defaultNamespace".

See: dapr/components-contrib#3060

### PR DESCRIPTION
**Pull request by mukundansundar**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Based on PR https://github.com/dapr/components-contrib/pull/3060/

## Issue reference

Closes #3669

### CODE REVIEW

**Comment by mukundansundar** [comment]
@ItalyPaleAle Please review PR

**Review by ItalyPaleAle — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-bindings/kubernetes-binding.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ spec:
 | `namespace` | Y | Input  | The Kubernetes namespace to read events from | `"default"` |
 | `resyncPeriodInSec` | N | Input | The period of time to refresh event list from Kubernetes API server. Defaults to `"10"` | `"15"`
 | `direction` | N | Input | The direction of the binding | `"input"`
+| `kubeconfigPath` | N | Input | The path to the kubeconfig file. If not specified, the binding will use the default in-cluster config value | `"/path/to/kubeconfig"`
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kubernetes-binding.md]:
```suggestion
| `kubeconfigPath` | N | Input | The path to the kubeconfig file. If not specified, the binding uses the default in-cluster config value | `"/path/to/kubeconfig"`
```

File: daprdocs/content/en/reference/components-reference/supported-cryptography/kubernetes-secrets.md [hunk 2]
```diff
@@ -33,7 +33,11 @@ The above example uses secrets as plain strings. It is recommended to use a secr
 
 ## Spec metadata fields
 
-For the Kubernetes secret store component, there are no metadata attributes.
+| Field              | Required |  Details | Example |
+|--------------------|:--------:|------------|-----|---------|
+| `defaultNamespace` | N | Default namespace to retrieve secrets from. If unset, the namespace must be specified for each key, as `namespace/secretName/key` | `"default-ns"` |
+| `kubeconfigPath` | N | The path to the kubeconfig file. If not specified, the binding will use the default in-cluster config value | `"/path/to/kubeconfig"`
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-cryptography/kubernetes-secrets.md]:
```suggestion
| `kubeconfigPath` | N | The path to the kubeconfig file. If not specified, the binding uses the default in-cluster config value | `"/path/to/kubeconfig"`
```

File: daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md [hunk 3]
```diff
@@ -32,7 +32,12 @@ spec:
 ```
 
 ## Spec metadata fields
-For the Kubernetes secret store component, there are no metadata attributes.
+
+| Field              | Required |  Details | Example |
+|--------------------|:--------:|------------|-----|---------|
+| `defaultNamespace` | N | Default namespace to retrieve secrets from | `"default-ns"` |
+| `kubeconfigPath` | N | The path to the kubeconfig file. If not specified, the binding will use the default in-cluster config value | `"/path/to/kubeconfig"`
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md]:
```suggestion
| `kubeconfigPath` | N | The path to the kubeconfig file. If not specified, the binding uses the default in-cluster config value | `"/path/to/kubeconfig"`
```

File: daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md [hunk 4]
```diff
@@ -32,7 +32,12 @@ spec:
 ```
 
 ## Spec metadata fields
-For the Kubernetes secret store component, there are no metadata attributes.
+
+| Field              | Required |  Details | Example |
+|--------------------|:--------:|------------|-----|---------|
+| `defaultNamespace` | N | Default namespace to retrieve secrets from | `"default-ns"` |
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md]: Any consequences if this value is not set?

Comment (mukundansundar) [reply daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md]: No, `defaultNamespace` is not a required parameter.

Comment (mukundansundar) [reply daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md]: namespace is a required parameter to get secrets .... 
either through request metadata, or though `NAMESPACE` env var. 
https://github.com/dapr/components-contrib/blob/master/secretstores/kubernetes/kubernetes.go#L112-L126
```
func (k *kubernetesSecretStore) getNamespaceFromMetadata(metadata map[string]string) (string, error) {
	if val, ok := metadata["namespace"]; ok && val != "" {
		return val, nil
	}

	if val := os.Getenv("NAMESPACE"); val != "" {
		return val, nil
	}

	if k.md.DefaultNamespace != "" {
		return k.md.DefaultNamespace, nil
	}

	return "", errors.New("namespace is missing on metadata and NAMESPACE env variable, and no default namespace is set")
}
```

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**